### PR TITLE
Refactor BackButton tests to RTL

### PR DESCRIPTION
**Pull request by noahtallen**
#### Proposed Changes

Refactors the BackButton tests to RTL. I also took the opportunity to simplify and improve a few parts of the test suite. See #70873

#### Testing Instructions

- CI should pass

### CODE REVIEW

File: packages/help-center/src/components/test/back-button.tsx [hunk 1]
```diff
@@ -16,40 +17,39 @@ jest.mock( 'react-router-dom', () => ( {
...
-		} );
+	afterEach( () => {
+		mockHistoryGoBack.mockClear();
+		mockHistoryPush.mockClear();
+	} );
+
+	it( 'navigates to the root when back to root is true', async () => {
+		render( <BackButton backToRoot /> );
+
+		const btn = screen.getByRole( 'button' );
```
Comment (tyxla) [inline packages/help-center/src/components/test/back-button.tsx]: There's no need to use a variable - we could inline it if we use it just once.

File: packages/help-center/src/components/test/back-button.tsx [hunk 2]
```diff
@@ -16,40 +17,39 @@ jest.mock( 'react-router-dom', () => ( {
...
+		await userEvent.click( btn );
```
Comment (tyxla) [inline packages/help-center/src/components/test/back-button.tsx]: It's generally recommended to run `user.setup()` first.

**Review by tyxla — APPROVED** [review]
Looks great, thanks! Just a couple of minor suggestions before we 🚢 .

File: packages/help-center/src/components/test/back-button.tsx (hunk 1 above)
Comment (tyxla) [reply packages/help-center/src/components/test/back-button.tsx]: Addressed in https://github.com/Automattic/wp-calypso/pull/70917/commits/37d6c25917cc9afa17b706586e8141a52f577425.

File: packages/help-center/src/components/test/back-button.tsx (hunk 2 above)
Comment (tyxla) [reply packages/help-center/src/components/test/back-button.tsx]: Done in https://github.com/Automattic/wp-calypso/pull/70917/commits/f20b73dc1ab23dce4ae5b187ed1a335424c65813

**Comment by tyxla** [comment]
Thanks, @noahtallen! Going to ship this for you. 🚀

**Comment by noahtallen** [comment]
Thanks @tyxla!

> It's generally recommended to run user.setup() first.

Could you explain this more? (What benefit does that add?)

**Comment by tyxla** [comment]
> Could you explain this more? (What benefit does that add?)

Sure thing! 

Running `userEvent.setup()` first is one step further into making interactions with our components resemble the real user behavior as close as possible. The idea is to set up additional workarounds to the current session in order to enable some APIs (like the [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard) for example) in our tests. It also allows us to run the tests within the same user session, meaning, all methods on the same instance share one input device state. This allows multiple events triggered one after another to behave in a predictable way, as close as possible to interactions by a real user.

Here's how this is described by the RTL docs:

> When users interact in the browser by e.g. pressing keyboard keys, they interact with a UI layer the browser shows to them. The browser then interprets this input, possibly alters the underlying DOM accordingly and dispatches [trusted](https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted) events.
>
> The UI layer and trusted events are not programmatically available.
>
> Therefore user-event has to apply workarounds and mock the UI layer to simulate user interactions like they would happen in the browser.
>
> The userEvent.setup() API applies these workarounds to the document and allows you to [configure](https://testing-library.com/docs/user-event/options) an "instance" of user-event.
>
> Methods on this instance share one input device state, e.g. which keys are pressed.
>
> This allows to write multiple consecutive interactions that behave just like the described interactions by a real user.

I believe the idea is that the workarounds would grow over time, as the need grows for support of additional APIs that aren't supported by jsdom.

Finally, the [docs](https://testing-library.com/docs/user-event/setup/#direct-apis) are pretty concrete about what is the recommended usage:

> Using the methods on the instances returned by userEvent.setup() is recommended.

Hope that makes sense!

**Pull request merged**